lib/uktest/README.md

```diff
@@ -0,0 +1,117 @@
+# Unikraft’s Testing Framework.
+
+## Overview
+
+In `uktest`, tests are organised hierarchically powered by the the lowest common denominator: the assertion.
+This organisation is inspired by [KUnit](https://kunit.dev/), the Linux Kernel's in-house testing system.
+For licensing reasons the Unikraft project cannot use this source code.
+However, by inspiration, we can organise the `uktest` library following a similar pattern:
+
+1. The assertion: it repersents the lowest common denominator of a test: some boolean operation which, when true, a single test passes.
```

Comment:
I would use a simple list with assertion, test case and test suite ad their one sentence definition.

Then, I would add a subsection (`###`) detailing each.